src/main/java/com/sciamus/contractanalyzer/domain/checks/queues/kafka/KafkaTransformationCheck.java

```diff
@@ -0,0 +1,164 @@
+package com.sciamus.contractanalyzer.domain.checks.queues.kafka;
+
+import com.sciamus.contractanalyzer.domain.checks.queues.kafka.config.KafkaConsumFactory;
+import com.sciamus.contractanalyzer.domain.checks.queues.kafka.config.KafkaProducFactory;
+import com.sciamus.contractanalyzer.domain.checks.reports.Report;
+import com.sciamus.contractanalyzer.domain.checks.reports.ReportBuilder;
+import com.sciamus.contractanalyzer.domain.checks.reports.ReportResults;
+import io.vavr.control.Try;
+import org.apache.kafka.clients.consumer.Consumer;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.springframework.kafka.core.KafkaTemplate;
+
+import java.time.Duration;
+import java.util.List;
+import java.util.Random;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class KafkaTransformationCheck implements KafkaCheck {
+    final String name = "KafkaTransformationCheck";
+
+    private final KafkaProducFactory kafkaProducFactory;
+    private final KafkaConsumFactory kafkaConsumFactory;
+    private ReportBuilder reportBuilder = new ReportBuilder();
+    private String checkUniqueKey;
```

Comment:
## warn

I don't think so we have established that Checks should be stateful..